packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -187,5 +193,50 @@ describe('<DataGrid /> - Pagination', () => {
       fireEvent.click(screen.getByRole('button', { name: /next page/i }));
       expect(getColumnValues()).to.deep.equal(['Nike 1']);
     });
+
+    describe('AutoPageSize', () => {
+      it('should always render the same amount of rows and fit the viewport', () => {
+        const TestCaseAutoPageSize = (
+          props: Partial<DataGridProps> & { nbRows?: number; nbCols?: number; height?: number },
+        ) => {
+          const data = useData(props.nbRows || 100, props.nbCols || 10);
+
+          return (
+            <div style={{ width: 300, height: props.height || 300 }}>
+              <DataGrid
+                columns={data.columns}
+                rows={data.rows}
+                autoPageSize
+                pagination
+                {...props}
+              />
+            </div>
+          );
+        };
+        const height = 780;
+        const footerHeight = 52;
+        const nbRows = 27;
+        const expectedViewportRowsLength = Math.floor(
+          (height - DEFAULT_GRID_OPTIONS.headerHeight - footerHeight) /
+            DEFAULT_GRID_OPTIONS.rowHeight,
+        );
+        render(<TestCaseAutoPageSize nbRows={nbRows} height={height} />);
+        let rows = document.querySelectorAll('.MuiDataGrid-viewport [role="row"]');
```

Comment:
Did you consider using `screen.getAllByRole('row')` instead? I mean, is `.MuiDataGrid-viewport` relevant in the test?
```suggestion
        let rows = screen.getAllByRole('row');
```

x the others

Reply:
no otherwise it also takes the column header